libraries/protocol/include/graphene/protocol/asset_ops.hpp

```diff
@@ -34,6 +34,8 @@ namespace graphene { namespace protocol {
       fc::optional<flat_set<account_id_type>> whitelist_market_fee_sharing;
       // After BSIP81 activation, taker_fee_percent is the taker fee
       fc::optional<uint16_t>                  taker_fee_percent;
+      // MERGE NOTE: BSIP-74 additions here
+      fc::optional<uint16_t> force_settle_fee_percent;  // BSIP-87
```

Comment:
Question: (@abitmore) — Have I made the right choice to add this new `force_settle_fee_percent` parameter inside the `additional_asset_options` structure?  As I’ve thought more about it, it has seemed to me it maybe should instead have been added to a new `additional_bitasset_options` structure, to become an extension to `bitasset_options` (instead of `asset_options`).  This new parameter (along with `margin_call_fee_ratio` from BSIP74) are bit-asset-specific, whereas the remaining existing parameters are generic to all coin types, I believe. 

Advantages would be: 

* Not storing parameters for asset types to which they do not apply.

* Owners would use the `asset_update_bitasset_operation` rather than `asset_update_operation` to change the parameters, which seems more contextually appropriate.

Disadvantages would be:

* Introducing a little bit more code complexity (new structures, extension types).  But, this seems reasonable to me.

* There is some precedent already — e.g. there are bitasset-specific flags and permissions which are stored in the `asset_options` even though they’d make more sense in `bitasset_options`.   

* An asset owner wishing to update all the relevant parameters would need to broadcast two separate operations, one to update the common fees and one to update the bitasset fees.

* It would require a little bit of refactoring of the code already written, but I don’t think we’ve dug ourselves too big of a hole thus far.

As mentioned, this also affects BSIP74.  When I last spoke with @jmjatlanta, (some days ago), he was supportive of moving the parameter to the new structure, assuming nothing has changed since then.

What are your thoughts? Is there a reason to keep it in the common options? Or should we move these ones over to bitasset options?

Reply:
Yes, I think it's better to put those options in `bitasset_options`. There was already some work done in https://github.com/bitshares/bitshares-core/pull/2157, but I didn't merge it so far because I hoped that your pulls got merge first then I rebase. But now it seems it's better that you work on top of it, so I'll merge it, after doen please rebase or merge. Thanks.

Reply:
Update: #2157 has been merged into `hardfork` branch, please merge `hardfork` into the development branch again. Thanks.